core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -58,27 +64,51 @@ public AtomicFileWriter(File f) throws IOException {
     }
```

Comment:
@batmat as discussed where you going to deprecate this?

Reply:
@jtnord not in this PR no. It is out of scope to me, so I can possibly file it separately, but I don't want to _risk_ triggering another serie of comments and fixes to do on something that is not central to this very change.